### ISSUE TITLE
401 errors for some models?

### ISSUE THREAD

**Issue by cliffrowley** [OwenTruong/civitdl#90]
I'm not sure why but some models refuse to download (even with an API key) and return a 401 error, despite being able to download them manually from the website directly.

For example:

```
---------

APIException (API Status Code 401):
         Downloading model from CivitAI failed for model id, 282171, and version id, 317633

---------
Retrying to download the current model...
Now downloading "Yellow summer dress"...
            - Model ID: 282171
            - Version ID: 317633

---------
```

I've tried waiting and retrying, even waiting a whole day just in case it's one of their weird proxy errors, but nada.

**Comment by OwenTruong** [comment]
Strange, when I use wget it is able to fetch the model.... I will try to investigate more about this.

`
wget https://civitai.com/api/download/models/317633?token={api_key} --content-disposition
`

**Comment by OwenTruong** [comment]
Ok I managed to figure out the reason why. It was because the program didn't expect civitai to send a 401 error for unauthorized models. The program expected civitai to redirect to a url with "reason=download-auth" in the title. This bug will be fixed in version 3.0.12 in a sec
